### PR TITLE
Plugin clnrest Onboarding

### PR DESCRIPTION
**Pull request by ShahanaFarooqui**

### CODE REVIEW

File: plugins/clnrest/clnrest.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    return options
+
+class CLNRestApplication(BaseApplication):
+    def __init__(self, app, options=None):
+        from utilities.shared import REST_PROTOCOL, REST_HOST, REST_PORT
+        self.application = app
+        self.options = options or {}
+        plugin.log(f"REST server running at {REST_PROTOCOL}://{REST_HOST}:{REST_PORT}", "info")
+        super().__init__()
```
Comment (vincenzopalazzo) [inline plugins/clnrest/clnrest.py:64]: `super().__init__()` as the convention is the first call that is done inside the constructor

File: plugins/clnrest/utilities/rpc_routes.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    @rpcns.response(500, "Server error")
+    def get(self):
+        """Get the list of all valid rpc methods"""
+        try:
+            help_response = call_rpc_method(plugin, "help", [])
+            html_content = process_help_response(help_response)
+            response = make_response(html_content)
+            response.headers["Content-Type"] = "text/html"
+            return response
+
```
Comment (vincenzopalazzo) [inline plugins/clnrest/utilities/rpc_routes.py:23]:
```suggestion
```

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: plugins/clnrest/clnrest.py (hunk 1 above)
Comment (ShahanaFarooqui) [reply plugins/clnrest/clnrest.py:64]: Adding this call first throws an error as "Error: 'CLNRestApplication' object has no attribute 'options'" and does not start the plugin at all. So I had to add options and application before I could initialise BaseApplication class.

File: plugins/clnrest/utilities/rpc_routes.py (hunk 2 above)
Comment (ShahanaFarooqui) [reply plugins/clnrest/utilities/rpc_routes.py:23]: Thank you. Checked and removed extra lines from all files.

File: plugins/clnrest/utilities/rpc_routes.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+@rpcns.route("/notifications")
+class NotificationsResource(Resource):
+    def get(self):
+        try:
+            def notifications_stream():
+                while True:
+                    from .rpc_plugin import queue
+                    yield queue.get()
+            return Response(stream_with_context(notifications_stream()), mimetype="text/event-stream")
+
```
[the remark this answers is not in the document]
Comment (ShahanaFarooqui) [reply plugins/clnrest/utilities/rpc_routes.py:69]: Thank you. Checked and removed extra lines from all files.

File: plugins/clnrest/utilities/rpc_routes.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+class ListMethodsResource(Resource):
+    @rpcns.response(200, "Success")
+    @rpcns.response(500, "Server error")
+    def get(self):
+        """Get the list of all valid rpc methods"""
+        try:
+            help_response = call_rpc_method(plugin, "help", [])
+            html_content = process_help_response(help_response)
+            response = make_response(html_content)
+            response.headers["Content-Type"] = "text/html"
```
[the remark this answers is not in the document]
Comment (ShahanaFarooqui) [reply plugins/clnrest/utilities/rpc_routes.py:21]: This `v1/list-methods` endpoint will not be required by an application as they can call `v1/help` and consume the response directly. 

This endpoint is useful for Swagger only to get the human readable list of available methods. But I understand the confusion and added more details to the endpoint to make it clear. The list will look like below in Swagger:

```
--------------------------------------------------------------------------------------------------------------------------------------------------------------

Command: addgossip message
Category: utility
Description: Inject gossip {message} into gossipd
Verbose: HELP! Please contribute a description for this json_command!

--------------------------------------------------------------------------------------------------------------------------------------------------------------
...
```

**Pull request closed without merging**